test/models/IllnessStartDateSchema.test.ts

```diff
@@ -0,0 +1,157 @@
+import { schema } from 'app/models/fields/Date.schema';
+import { SchemaValidator } from 'app/utils/validation/SchemaValidator';
+import { ValidationError } from 'app/utils/validation/ValidationError';
+
+import { assertValidationErrors } from 'test/models/ValidationAssertions';
+
+const validator = new SchemaValidator(schema);
+
+const dayField: string = 'day';
+const monthField: string = 'month';
+const yearField: string = 'year';
+const startDateField: string = 'illnessStart';
+
+describe('IllnessStartDate schema', () => {
+
+    it('should reject empty object', () => {
+        const validationResult = validator.validate({});
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
```

Comment:
This is bit confusing of course and you work around it in the validator but maybe in the future we could try conditional schema so that either three ingredient fields are returned or one for full date.